server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -886,6 +886,7 @@ public UserVm resetVMSSHKey(ResetVMSSHKeyCmd cmd) throws ResourceUnavailableExce
             throw new CloudRuntimeException("Failed to reset SSH Key for the virtual machine ");
         }
 
+        _vmDao.loadDetails(userVm);
```

Comment:
I would prefer removing this line and including it into the `removeEncryptedPasswordFromUserVmVoDetails` method if the VM details are empty

Reply:
@nvazquez done. 
details are loaded but not updated, therefore it does not contain the new ssh key.